### PR TITLE
Add RDMA netlink socket for RDMA device information

### PR DESCRIPTION
**Pull request by paravmellanox**
This patch adds very basic support for getting information about RDMA
networking device; starting with device index, name, firmware version,
node GUID and system image GUID.
This is done through RDMA netlink socket.

RDMA devices are some what similar to Ethernet devices.
However there are few major differences between them.
RDMA devices usually have one or two ports, unlike Ethernet devices.
Each port has its own attributes, state and network addresses which are
different than Ethernet devices (Link and LinkAttrs). They almost don't
overlap with Link and LinkAttrs.

Therefore it doesn't derive Link and LinkAttrs structure; instead they
are represented using RdmaLink and RdmaLinkAttrs.

RdmaLink represents a RDMA device containing its attributes.
All Rdma device communication occurs through rdma subsystem's netlink
socket.

Signed-off-by: Parav Pandit parav@mellanox.com

### CODE REVIEW

File: rdma_link_linux.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			return nil, err
+		}
+		if link.Attrs.Name == name {
+			return link, nil
+		}
+	}
+	return nil, fmt.Errorf("Rdma device %v not found", name)
+}
+
+// LinkByName finds a link by name and returns a pointer to the object.
```
Comment (aboch) [inline rdma_link_linux.go]: Given this is exported, pls fix the function name in the comment. Or just remove the comment, I don't think it is actually needed.

File: rdma_link_test.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	"strings"
+	"testing"
+)
+
+func setupRdmaKModule(t *testing.T, name string) {
+	file, err := ioutil.ReadFile("/proc/modules")
+	if err != nil {
+		t.Fatal("Failed to open /proc/modules", err)
+	}
+	found := false
```
Comment (aboch) [inline rdma_link_test.go]: You could drop this variable:
```
[...]
       if n == name {
              t.SkipUnlessRoot();
              return;
        }
   }
   t.Skipf("Test requires kmodule %q.", name)
}
```

**Comment by aboch** [comment]
Thanks @paravmellanox . Just a minor comment.

File: rdma_link_linux.go (hunk 1 above)
Comment (paravmellanox) [reply rdma_link_linux.go]: @aboch, I will fix the comment.

File: rdma_link_test.go (hunk 2 above)
Comment (paravmellanox) [reply rdma_link_test.go]: @aboch, sure. I will simplify this.

File: rdma_link_linux.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+func RdmaLinkByName(name string) (*RdmaLink, error) {
+	return pkgHandle.RdmaLinkByName(name)
+}
+
+func (h *Handle) RdmaLinkByName(name string) (*RdmaLink, error) {
```
Comment (fcrisciani) [inline rdma_link_linux.go]: missing comment for exported method

Comment (paravmellanox) [reply rdma_link_linux.go]: @fcrisciani right; adding it.

**Comment by aboch** [comment]
Looks like we do not run golint in this project, otherwise CI would have pointed out when exported function comments do not start with the function name.

If we ever decide to enable it in future, at that time we can take care of this issue across all files at once.

LGTM

**Comment by paravmellanox** [comment]
I added the comment for exported function, addressed comments and pushed the changes again.

**Pull request merged**